File: benchmarks/comparison/integer-recurrence.ts
import { inputInteger, logInteger } from "./host";
function kernel(n: i32, seed: i32): i32 {
    let state = seed;
    for (let i = 0; i < n; ++i) state = (state * 17 + 23) % 65521;
    return state;
}
export function run(): void {
    const n = inputInteger();
    const seed = inputInteger();
    logInteger(kernel(n, seed));
}

File: benchmarks/comparison/integer-recurrence.c
#include "host.h"
static int kernel(int n, int seed) {
    int state = seed;
    for (int i = 0; i < n; ++i) state = (state * 17 + 23) % 65521;
    return state;
}
void run(void) {
    int n = inputInteger();
    int seed = inputInteger();
    logInteger(kernel(n, seed));
}
